cmd/kpod/login.go

```diff
@@ -64,21 +78,34 @@ func loginCmd(c *cli.Context) error {
 	if err != nil {
 		return errors.Wrapf(err, "error getting username and password")
 	}
+	sc.DockerInsecureSkipTLSVerify = !c.BoolT("tls-verify")
 
-	if err = docker.CheckAuth(context.TODO(), sc, username, password, server); err == nil {
-		if err := config.SetAuthentication(sc, server, username, password); err != nil {
-			return err
-		}
+	var certDirs []string
+	if c.String("cert-dir") != "" {
+		certDirs = []string{c.String("cert-dir")}
+	} else {
+		certDirs = []string{CertsDir, "/etc/docker/certs.d/"}
 	}
-	switch err {
-	case nil:
-		fmt.Println("Login Succeeded!")
-		return nil
-	case docker.ErrUnauthorizedForCredentials:
-		return errors.Errorf("error logging into %q: invalid username/password\n", server)
-	default:
-		return errors.Wrapf(err, "error authenticating creds for %q", server)
+
+	for _, dir := range certDirs {
+		// set DockerCert Path
+		sc.DockerCertPath = filepath.Join(dir, server)
+
+		if err = docker.CheckAuth(context.TODO(), sc, username, password, server); err == nil {
+			if err := config.SetAuthentication(sc, server, username, password); err != nil {
+				return err
+			}
+		}
+		switch err {
+		case nil:
+			fmt.Println("Login Succeeded!")
+			return nil
+		case docker.ErrUnauthorizedForCredentials:
+			return errors.Errorf("error logging into %q: invalid username/password\n", server)
```

Comment:
Should we be returning the error and leaving the loop while there are still directories to check for certs?

Reply:
I don't think we'd hit it often if at all, but to be ultra safe it would probably be good to have a 'default:' statement with another Errorf with some kind of "errors.Wrapf(err, "Error logging into server")" or some such.

Reply:
I left it as it is because, this error is returned only when creds are wrong! While we are looping to find the location of the certs, so the thought was if the location is found but creds are wrong we should error out.

Reply:
@TomSweeneyRedHat we can only have default case with a warning, having a default with return might not be helpful.

Reply:
Good point, if we don't have a cert-dir specified by the user, we need to check the default OCI location (CertsDir) then if that fails, check the Docker default location.  So we shouldn't return until we've tried both.

Reply:
Hm. I think that should be fine, then.

Reply:
Summing.  I don't think we should fail immediately if the creds are wrong.  For instance if there's OCI certs with one un/pwd and Docker certs with a separate un/pwd and the Docker un/pwd is given.  We'd fail on the OCI certs and return out when I think we'd want to then try the Docker creds.

Granted an unlikely scenario.   I'm going to guess that 99.9999999% (and probably a few more 9's) of the people will only have certs in one directory.

Reply:
@TomSweeneyRedHat made changes as you have suggested.